src/components/source_editor/LatticeConfigurationDialog.js

```diff
@@ -108,10 +109,34 @@ class LatticeConfigurationDialog extends React.Component {
         const newMaterial = new Material(newMaterialConfig);
         // assert basis is stored in 'crystal' units
         newMaterial.toCrystal();
+        newMaterial.isNonPeriodic = this.state.isNonPeriodic;
+
         this.props.onUpdate(newMaterial);
         this.props.onSubmit();
     }
 
+    renderNonPeriodicSwitch() {
+        return (
+            <div className="col-xs-12 p-0">
+                <div className="col-xs-6 p-25">
+                    <ToggleSwitch
+                        color="blue" title="Non-Periodic"
+                        class="pull-left"
+                        onStateChange={() => this.setState({isNonPeriodic: !this.state.isNonPeriodic})}
+                        checked={this.state.isNonPeriodic}
+                        id="access-level"
+                    />
+                </div>
+                <div className="col-xs-6">
+                    <button className="btn btn-custom pull-right save-lattice-config" data-dismiss="modal"
+                            onClick={this.handleUpdateLattice}>
+                        {this.props.submitButtonTxt || "Apply Non-Periodic"}
```

Comment:
Do we need a separate submit button or just the toggle switch itself would be enough?

Reply:
I was thinking we could go either or. We also could use the the `apply edits` button instead to apply the lattice changes instead of a separate `non-periodic` button